src/main/java/automatabuilder/DFA.java

```diff
@@ -58,7 +60,51 @@ public boolean test(String word) throws InvalidWordException {
         
         return end.isFinal();
     }
-    
+
+    @Override
+    public String testFile(File testFile) {
+        int maxLength = 20;
+        StringBuilder result = new StringBuilder();
+        Scanner scanner = null;
+        try {
+            scanner = new Scanner(testFile);
+        } catch (FileNotFoundException e) {
+            System.err.println("Error trying to read test file");
+            return "";
+        }
+
+        while (scanner.hasNext()) {
+            String word = scanner.nextLine();
+            StringBuilder line = new StringBuilder();
+            String suffix;
+            boolean accept;
+            try {
+                accept = test(word);
+                suffix = accept ? "Accepted" : "Rejected";
+            } catch (InvalidWordException e) {
+                suffix = "Invalid word";
+            }
+            if (word.length() > maxLength) {
+                for (int i = 0; i < maxLength-3; i++) {
+                    line.append(word.charAt(i));
+                }
+                line.append(".. ");
+            }
+            else {
+                line.append(word);
+                for (int i = line.length(); i < maxLength; i++) {
+                    line.append(" ");
+                }
+            }
+            line.append(suffix);
+            line.append("\n");
+            result.append(line.toString());
+            //set the stringbuilder to the empty string
+            line.setLength(0);
+        }
+        return result.toString();
```

Comment:
Key as the input string and the value denoting if the string was accepted. The problem here is that it doesn't support "Invalid word" vs "Not accepted", but im not sure thats important :P